pkg/migrations/duplicate.go

```diff
@@ -171,16 +182,30 @@ func (d *Duplicator) Duplicate(ctx context.Context) error {
 		}
 
 		if slices.Contains(uc.Columns, d.column.Name) {
+			cols := uc.Columns
+			// Drop the existing unique index if it is a duplicated unique constraint
+			if existingConstraint, ok := d.duplicatedUnique[DuplicationName(uc.Name)]; ok {
+				dropIndex := fmt.Sprintf("DROP INDEX CONCURRENTLY %s", pq.QuoteIdentifier(DuplicationName(uc.Name)))
+				_, err := d.conn.ExecContext(ctx, dropIndex)
+				if err != nil {
+					return err
+				}
```

Comment:
I'm not sure that creating and then dropping indexes duplicated by previous columns is the way to go here.

Previously the column duplicator was stateless; it duplicated columns without reference to other columns that may have been duplicated previously. The need to duplicate multi-column constraints means it has to become stateful to some extent.

Some options for how to handle column duplication now that it needs to be stateful:

* Introduce some kind of duplication context type that can hold state between calls to the duplicator.
* Redesign the duplicator to be a multi-column duplicator.

Not sure OTOH which is better (maybe the second option?) but I think we need to accept that column duplication is now stateful in the presence of multi-column constraints and design it accordingly.

Reply:
I already implemented the multi-column duplicator. However, i decided to submit this as it was a smaller change. Anyway the multi-column duplicator is probably a better approach as i need to adjust the duplication of check constraints as well.